core/src/main/scala/io/chrisdavenport/npmpackage/NpmConfig.scala

```diff
@@ -31,16 +29,34 @@ object NpmConfig {
     targetFile
   }
 
+  // @myscope:registry=https://mycustomregistry.example.org/:_authToken=\\${NPM_TOKEN}
+  // So this file is constructed by a set of registrys defined for scopes.
+  // And for you to be able to set custom kv pairs for any of the defined registry urls.
+
+  //
+  // registries
+  // This is necessary for any custom scope dependency resolutions.
+  // scope -> registry *
+  // keys
+  // registry -> key -> value *
+
+  def customRegistry(scope: String, registry: String): String =
+    s"$scope:registry=$registry"
```

Comment:
Since we're adding the `@` on NpmPackagePlugin.scala:194, I think we need to do so here, too. Otherwise we could omit it from both places and expect users to add it to `npmPackageNpmrcScope`